File: examples/typescript/estimate.ts
const baseUrl = process.env.AQUASTAT_BASE_URL || "http://localhost:8080";
const apiKey = process.env.AQUASTAT_API_KEY;

if (!apiKey) {
  throw new Error("AQUASTAT_API_KEY is required");
}

async function main(): Promise<void> {
  const response = await fetch(`${baseUrl}/api/v1/estimate?provider=aws&region=us-east-1&load_mw=2.5`, {
    headers: { "X-API-Key": apiKey }
  });

  if (!response.ok) {
    throw new Error(await response.text());
  }

  console.log(await response.json());
}

void main();

File: examples/python/estimate.py
import os

import httpx


def main() -> None:
    api_key = os.environ["AQUASTAT_API_KEY"]
    base_url = os.environ.get("AQUASTAT_BASE_URL", "http://localhost:8080")
    response = httpx.get(
        f"{base_url}/api/v1/estimate",
        params={"provider": "aws", "region": "us-east-1", "load_mw": 2.5},
        headers={"X-API-Key": api_key},
        timeout=10.0,
    )
    response.raise_for_status()
    print(response.json())


if __name__ == "__main__":
    main()
